File: leetcode/2001-2999/2390-removing-stars-from-a-string/RemovingStarsFromAString.ts
function removeStars(s: string): string {
  let result: string[] = [];

  for (const c of s.split("")) {
    if (c === '*') {
      result.pop();
    } else {
      result.push(c);
    }
  }

  return result.join("");
};

console.log(removeStars("leet**cod*e"));

File: leetcode/2001-3000/2301-2400/2390-removing-stars-from-a-string/RemovingStarsFromAString.java
public class RemovingStarsFromAString {
  public static void main(String[] args) {
    System.out.println(new Solution().removeStars("leet**cod*e"));
  }
}

class Solution {
  public String removeStars(String s) {
    StringBuilder sb = new StringBuilder();

    for (int i = 0; i < s.length(); i++) {
      if (s.charAt(i) == '*') {
        if (!sb.isEmpty()) {
          sb.deleteCharAt(sb.length() - 1);
        }
      } else {
        sb.append(s.charAt(i));
      }
    }

    return sb.toString();
  }
}
